aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -0,0 +1,104 @@
+#include "ATen/NativeFunctions.h"
+#include "ATen/Dispatch.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include <curand.h>
+#include <curand_kernel.h>
+#include <curand_philox4x32_x.h>
+#include <thrust/functional.h>
+
+#include <THC/THCGeneral.h>
+#include <THC/THCHalf.h>
+#include <THC/THCApply.cuh>
+#include <THC/THCNumerics.cuh>
+#include <THC/THCTensorRandom.h>
+
+THCGenerator* THCRandom_getGenerator(THCState* state);
+
+namespace at {
+namespace native {
+
+namespace dist {
+  thrust::pair<uint64_t, uint64_t> get_philox_seed(Generator *gen) {
+    auto gen_ = THCRandom_getGenerator(at::globalContext().thc_state);
+    return thrust::make_pair(gen_->initial_seed, gen_->philox_seed_offset++);
```

Comment:
Ah, `thrust::pair` isn't thread-safe? I can definitely make it `std::pair`. I think we do have a per-GPU seed, based on https://github.com/pytorch/pytorch/blob/master/aten/src/THC/THCTensorRandom.h#L18, but since I don't know `thrust`'s thread-safety, I'll just make it `std::pair`.